src/server/pkg/obj/amazon_client.go

```diff
@@ -63,15 +74,45 @@ func (c *amazonClient) Reader(name string, offset uint64, size uint64) (io.ReadC
 		byteRange = fmt.Sprintf("bytes=%s", byteRange)
 	}
 
-	getObjectOutput, err := c.s3.GetObject(&s3.GetObjectInput{
-		Bucket: aws.String(c.bucket),
-		Key:    aws.String(name),
-		Range:  aws.String(byteRange),
-	})
-	if err != nil {
-		return nil, err
+	var reader io.ReadCloser
+	if c.usingCloudfront() {
+		var resp *http.Response
+		var connErr error
+		url := fmt.Sprintf("http://%v.cloudfront.net/%v", c.distribution, name)
+
+		req, err := http.NewRequest("GET", url, nil)
+		if err != nil {
+			return nil, err
+		}
+		req.Header.Add("Range", byteRange)
+
+		backoff.RetryNotify(func() error {
+			resp, connErr = http.DefaultClient.Do(req)
+			if connErr != nil && isNetRetryable(connErr) {
+				return connErr
+			}
```

Comment:
It seems a little weird to me that we swallow no retryable errors here, I'm guessing we'll recover most of them when `resp.StatusCode >= 300` but I'm worried there might be a few that don't get reported here and don't get caught below either. You can handle stuff like this the `Notify` function below which I think might be a bit more idiomatic. I.e. in your `Notify` function you'd check if the error is `isNetRetryable` if it is you return `nil`, otherwise you return the error. Seems like a slightly safer way to handle the errors.